File: thsoft-gen/src/main/java/com/thsoft/metamodel/ui/script/VueDeclareScriptBlock.java
package com.thsoft.metamodel.ui.script;

import java.util.ArrayList;
import java.util.List;

public class VueDeclareScriptBlock extends ScriptBlock {

	private List<String> datas = new ArrayList<String>();
	private List<VueMethodScriptBlock> methods = new ArrayList<VueMethodScriptBlock>();
	
	public VueDeclareScriptBlock(List<String> vueData) {
		this.setDatas(vueData);
	}

	public List<String> getDatas() {
		return datas;
	}

	public void setDatas(List<String> datas) {
		this.datas = datas;
	}

	public List<VueMethodScriptBlock> getMethods() {
		return methods;
	}

	public void setMethods(List<VueMethodScriptBlock> methods) {
		this.methods = methods;
	}

}

File: thsoft-gen/code/src/main/resources/template/js/vueDeclareScriptBlock.js
@args ClientScript clientScript, VueDeclareScriptBlock scriptBlock
@import com.thsoft.metamodel.ui.*
@import com.thsoft.metamodel.ui.script.*
@raw() {
	var vueObject = new Vue({
    	el:'#vueData',
        data:function(){
            return {
            @for(VueData vueData : scriptBlock.getDatas()){
				@if (vueData.getDefaultValue() == null) {
@vueData.getName(): {},
				}
				else {
@vueData.getName(): @vueData.getDefaultValue(),
				}
            }
            }
        },
        methods:{
			@for(VueMethodScriptBlock vueMethod : scriptBlock.getMethods()){
				@vueMethod.render()
            }
        }
	});
}